File: sdks/javascript/src/runtime/run-result.ts
import type {Usage} from "../model/model-response.js";
import type {RunEvent} from "./run-event.js";

/** Successful terminal value and complete observed event trace. */
export interface RunResult {

  /** Stable identifier for the completed run. */
  readonly runId: string;

  /** Final assistant text. */
  readonly output: string;

  /** Aggregate normalized token usage. */
  readonly usage: Usage;

  /** Number of completed model turns. */
  readonly turns: number;

  /** Every event observed before the terminal result. */
  readonly events: readonly RunEvent[];

}

File: sdks/kotlin/src/main/kotlin/dev/orion/sdk/runtime/RunResult.kt
package dev.orion.sdk.runtime

import dev.orion.sdk.model.Usage

/** Terminal successful value returned by a run. */
internal data class RunResult(
    /** Identifier of the completed run. */
    public val runId: String,
    /** Final assistant text. */
    public val output: String,
    /** Aggregate normalized token usage. */
    public val usage: Usage,
    /** Number of completed model turns. */
    public val turns: Int,
    /** Immutable event trace in sequence order. */
    public val events: List<RunEvent>,
) : RunItem
